curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657c7e53fa2354011942a466.md

```diff
@@ -0,0 +1,291 @@
+---
+id: 657c7e53fa2354011942a466
+title: Step 11
+challengeType: 0
+dashedName: step-11
+---
+
+# --description--
+
+Every time the user clicks on the rules button, the current boolean value for the `isModalShowing` variable should toggle between `true` and `false`. 
+
+Use the `logical NOT operator(!)` in front of your `isModalShowing` variable to invert the value and reassign that to the `isModalShowing` variable. 
+
+# --hints--
+
+You should use the `logical NOT operator(!)` in front of your `isModalShowing` variable.
+
+```js
+assert.match(code, /!\s*isModalShowing/);
+```
+
+You should reassign the inverted `isModalShowing` value to `isModalShowing`.
+
+```js
+assert.match(code, /isModalShowing\s*=\s*!\s*isModalShowing/);
```

Comment:
A bit of an edge case, but this should prevent people from passing with `const isModalShowing = !isModalShowing`:

```suggestion
assert.match(code, /^\s*isModalShowing\s*=\s*!\s*isModalShowing/m);
```